### ISSUE TITLE
Docs: Add post on parserOptions.project = true

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6520]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

As part of writing #6515, I wrote up some docs on using `parserOptions.project = true` (#6084). That feature isn't actually part of v6 - it's just something I'd wanted to get in before or in v6.

Since it's already landed in v5, let's add a blog post explaining it! It'll be good visibility for the enhancement.

<details>
<summary>Here's what I originally drafted...</summary>

### Feature: Relative `tsconfig.json` Projects

In typescript-eslint v5, in order to enable [typed linting](https://typescript-eslint.io/linting/typed-linting), your ESLint configuration must include a `parserOptions.project` property indicating which TSConfig(s) to use:

```js title=".eslintrc.cjs"
module.exports = {
  // ...
  parserOptions: {
    project: ['./tsconfig.json'],
  },
  // ...
};
```

This works well for repositories that have a single TSConfig for all files, such as small repositories using a single `tsconfig.json` or larger monorepos that have a `tsconfig.eslint.json` specifically used for linting.
But what if your project uses different TSConfig options for different TypeScript files in nested subdirectories?
Specifying the right array of relative TSConfig paths to try for each file in order can be cumbersome or downright impossible.

We recently added the option to specify `parserOptions.project` as just `true` in ESLint configurations.
This indicates to typescript-eslint that each file should be linted with the closest `tsconfig.json` file on disk.

```js title=".eslintrc.cjs"
module.exports = {
  // ...
  parserOptions: {
    project: true,
  },
  // ...
};
```

We hope using `project = true` simplifies configurations for repositories with nontrivial ESLint configuration setups.
We intend to add additional options around automatic TSConfig lookups to allow additional file names or other convenient behaviors.
</details>

### Affected URL(s)

https://typescript-eslint.io/blog/relative-parser-options-projects perhaps? Something like that?

**Comment by onionhammer** [comment]
I was just about to ask about tsconfig relative to eslint file; in what version is this available?

**Comment by JoshuaKGoldberg** [comment]
It's already available today! https://typescript-eslint.io/linting/typed-linting#specifying-tsconfigs.

We release on Mondays, and #6084 was merged into `main` on Thursday Feb 9, so all versions since 5.52.0: https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.52.0

Btw, per https://typescript-eslint.io/contributing/issues#commenting, we ask that you ask questions like separately from the issue tracker. We _just_ opened up our Discord! 🙌
